packages/dialog/src/vaadin-dialog.js

```diff
@@ -269,9 +269,15 @@ class Dialog extends OverlayClassMixin(
   /** @protected */
   disconnectedCallback() {
     super.disconnectedCallback();
-    // Close overlay and memorize opened state
-    this.__restoreOpened = this.opened;
-    this.opened = false;
+    // Automatically close the overlay when dialog is removed from DOM
+    // Using a timeout to avoid toggling opened state, and dispatching change
+    // events, when just moving the dialog in the DOM
+    setTimeout(() => {
```

Comment:
Do we have other tests possibly affected by this change other than `confirm-dialog.test.js`?

Reply:
I'm not 100% sure what it could impact. The only obvious difference was that when running a test suite there could now be multiple overlays open, which affected the `z-index` style of sub-sequent overlays.